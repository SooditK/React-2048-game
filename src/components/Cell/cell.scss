@import '../../assets/styles/vars';

$web-len: 6vw;
$mobile-len: 20vw;

.cell {
  align-items: center;
  background-color: $black;
  border-radius: 5px;
  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.18), 0 1px 5px 0 rgba(0, 0, 0, 0.15);
  display: flex;
  height: $web-len;
  justify-content: center;
  width: $web-len;
}

@media all and (max-width: $break-point) {
  .cell {
    height: $mobile-len;
    width: $mobile-len;
  }
}

.number {
  color: $white;
  font-size: 2rem;
}

.color-2 {
  background-color: $blue;
}

.color-4 {
  background-color: $green;
}

.color-8 {
  background-color: $purple;
}

.color-16 {
  background-color: $red;
}

.color-32 {
  background-color: $pink;
}

.color-64 {
  background-color: $lime;
}

.color-128 {
  background-color: $indigo;
}

.color-256 {
  background-color: $orange;
}

.color-512 {
  background-color: $deep-purple;
}

.color-1024 {
  background-color: $teal;
}

.color-2048 {
  background-color: $light-blue;
}

.color-4096 {
  background-color: $yellow;
}

.color-8192 {
  background-color: $cyan;
}
