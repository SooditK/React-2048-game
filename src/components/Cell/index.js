import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './cell.scss';
import { isObjEqual } from '../../utils/helpers';

// Game board cell, minimum component in game board,
// one component stand for a number.
export default class Cell extends React.Component {
  static propTypes = {
    value: PropTypes.number.isRequired
  };

  shouldComponentUpdate(nextProps, nextState) {
    return (
      !isObjEqual(nextProps, this.props) || !isObjEqual(nextState, this.state)
    );
  }

  render() {
    const {
      props: { value }
    } = this;

    const color = `color-${value}`;
    return (
      <td>
        <div
          className={classnames([styles.cell, { [styles[color]]: !!value }])}
        >
          <div className={styles.number}>{value || null}</div>
        </div>
      </td>
    );
  }
}
