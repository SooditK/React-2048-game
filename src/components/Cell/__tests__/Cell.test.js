import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Cell from '..';

Enzyme.configure({ adapter: new Adapter() });

describe('<Board />', () => {
  it('component render', () => {
    let cell = renderer.create(<Cell value={32} />).toJSON();
    expect(cell).toMatchSnapshot();

    cell = renderer.create(<Cell value={2048} />).toJSON();
    expect(cell).toMatchSnapshot();
  });

  it('shouldComponentUpdate', () => {
    const tr = document.createElement('tr');
    const cell = mount(<Cell value={32} />, {
      attachTo: tr
    });
    expect(cell.props().value).toEqual(32);
    let shouldUpdate = cell.instance().shouldComponentUpdate(
      {
        value: 32
      },
      null
    );
    expect(shouldUpdate).toBe(false);
    shouldUpdate = cell.instance().shouldComponentUpdate(
      {
        value: 233
      },
      null
    );
    expect(shouldUpdate).toBe(true);
  });
});
