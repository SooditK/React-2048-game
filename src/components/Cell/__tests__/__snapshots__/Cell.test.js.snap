// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> component render 1`] = `
<td>
  <div
    className="cell color-32"
  >
    <div
      className="number"
    >
      32
    </div>
  </div>
</td>
`;

exports[`<Board /> component render 2`] = `
<td>
  <div
    className="cell color-2048"
  >
    <div
      className="number"
    >
      2048
    </div>
  </div>
</td>
`;
